### PR TITLE
[bitnami/mongodb] feat: Add network policy template into mongodb chart

### PR DESCRIPTION
**Pull request by kovasaurus**
### Description of the change

Adding the network policy template into mongodb chart. 

### Benefits

Allowing the chart to utilize the native Kubernetes Network policies.

### Applicable issues

- fixes #

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/mongodb/values.yaml [hunk 1]
```diff
@@ -1002,6 +1002,57 @@ externalAccess:
...
+  ## @param networkPolicy.enabled Enable MongoDB(&reg;) network policies
+  ##
+  enabled: false
+
+  ## Ingress rules - Allow incoming connections
+  ##
+  ingress:
+    ## @param networkPolicy.ingress.namespaceSelector [object] Namespace selector label that is allowed to access the MongoDB node. This label will be used to identify the allowed namespace(s).
+    ## @param networkPolicy.ingress.podSelector [object] Pod selector label that is allowed to access the MongoDB node. This label will be used to identify the allowed pod(s).
+    ## @param networkPolicy.ingress.customRules [array] Custom network policy for the MongoDB node.
```
Comment (andresbono) [inline bitnami/mongodb/values.yaml:1021]: Can you update the application name references as you did for `networkPolicy.enabled`?

```suggestion
    ## @param networkPolicy.ingress.namespaceSelector [object] Namespace selector label that is allowed to access the MongoDB(&reg;) node. This label will be used to identify the allowed namespace(s).
    ## @param networkPolicy.ingress.podSelector [object] Pod selector label that is allowed to access the MongoDB(&reg;) node. This label will be used to identify the allowed pod(s).
    ## @param networkPolicy.ingress.customRules [array] Custom network policy for the MongoDB(&reg;) node.
```

File: bitnami/mongodb/values.yaml [hunk 2]
```diff
@@ -1002,6 +1002,57 @@ externalAccess:
...
+    ##     - namespaceSelector:
+    ##         matchLabels:
+    ##            label1: value1
+    ##       podSelector:
+    ##         matchLabels:
+    ##            label2: value2
+  ## Egress rules - deny outgoing connections
+  ##
+  egress:
+    ## @param networkPolicy.egress.customRules [array] Custom network policy for the MongoDB node.
```
Comment (andresbono) [inline bitnami/mongodb/values.yaml:1043]:
```suggestion
    ## @param networkPolicy.egress.customRules [array] Custom network policy for the MongoDB(&reg;) node.
```

File: bitnami/mongodb/templates/network-policy.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+Copyright VMware, Inc.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{- if and .Values.networkPolicy.enabled }}
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ include "mongodb.fullname" . }}
+  namespace: {{ include "mongodb.namespace" . | quote }}
```
Comment (andresbono) [inline bitnami/mongodb/templates/network-policy.yaml:11]: Are these changes based on some other existing chart? Did you check any of the existing `networkpolicy.yaml` templates in particular? I would suggest that you check a recent one: bitnami/mlflow/templates/tracking/networkpolicy.yaml that should include all the best practices. Just as an example:

```suggestion
  name: {{ template "mongodb.fullname" . }}
  namespace: {{ include "common.names.namespace" . | quote }}
```

**Review by andresbono — COMMENTED** [review]
Thank you for contributing with this PR! Could you please check my comments?

File: bitnami/mongodb/templates/network-policy.yaml (hunk 3 above)
Comment (kovasaurus) [reply bitnami/mongodb/templates/network-policy.yaml:11]: Yea i was checking out the postgres and redis `networkpolicy.yaml` templates which used .Release.Namespace and some other templates in mongoDB used the mongodb.namespace syntax so i went with it to have the consistency with mongo chart. However I can change it to the proposed one

File: bitnami/mongodb/values.yaml (hunk 1 above)
Comment (andresbono) [reply bitnami/mongodb/values.yaml:1022]: I think this suggestion was missed. Does it make sense to apply it?

File: bitnami/mongodb/templates/networkpolicy.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
+{{- /*
+Copyright VMware, Inc.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{- if .Values.networkPolicy.enabled }}
+apiVersion: networking.k8s.io/v1
```
Comment (andresbono) [inline bitnami/mongodb/templates/networkpolicy.yaml:7]: What if we use this common helper to set the apiVersion?
```suggestion
apiVersion: {{ include "common.capabilities.networkPolicy.apiVersion" . }}
```

File: bitnami/mongodb/values.yaml (hunk 2 above)
Comment (andresbono) [reply bitnami/mongodb/values.yaml:1044]: What about this last suggestion? Thanks!

**Review by andresbono — APPROVED** [review]
LGTM, thank you for your contribution!

**Pull request merged**